site/content/guidelines/approach.md

```diff
@@ -0,0 +1,36 @@
+# How to Have Constructive Code Review Conversations
+
+> adapted from a [guide](http://carver.cs.ua.edu/Slides/2019/URSSI-WinterSchool/URSSI-WinterSchool-PeerCodeReview.pdf) by [Jeff Carver](http://carver.cs.ua.edu)
+
+Many aspects of code review can occur asynchronously,
+if the software is mature and the scope of the project is clearly defined.
+Unfortunately, when developing code for research,
+the software is not mature and the scope is not well understood,
+or it wouldn't be research.
```

Comment:
I have mixed feelings about the motivation for face-to-face code reviews here, although I'm struggling to formulate an alternative. I'm okay to merge this and I'll make a note - maybe something worth discussing in an upcoming meeting :)

I think I get the point here, but many researchers would argue their software is mature and the scope is well understood. To be effective and efficient, asynchronous review requires both infrastructure, organisation and levels of commitment to code quality unusual in research context. Therefore face-to-face code reviews are easier? That's not to say that face-to-face code reviews are not a lesser alternative to asynchronous reviews - for instance they make the context and author's mindset easier to grasp for a reviewer.

Reply:
> I think I get the point here, but many researchers would argue their software is mature and the scope is well understood. 

I agree but I think the huge gap in the literature is "how do you do review when software is **not** mature and the scope is **not** well understood. With the exception of Petre Wilson 2015

If I want a guide for code review when software is mature and scope is well understood, I can already find that

Reply:
Okay, that's a good point :)